'SpanContext' is not a member of 'opentelemetry::v1::sdk::trace'

Hello! I am trying to add the OpenTelemetry SDK as a dependency into our project, and I am consistently hitting this compile issue. I would appreciate some help.

**Describe your environment** Describe any aspect of your environment relevant to the problem, including your platform, build system, version numbers of installed dependencies, etc. If you're reporting a problem with a specific version of a library in this repo, please check whether the problem has been fixed on main branch.

* CentOS in Parallel 18
* The latest release `opentelemetry-cpp-1.12.0.tar.gz` downloaded

**Steps to reproduce**
Describe exactly how to reproduce the error. Include a code sample if applicable.

 When I include certain dependencies from metrics SDK, for example any of the following
```
#include <opentelemetry/sdk/metrics/export/periodic_exporting_metric_reader_factory.h>
#include <opentelemetry/sdk/metrics/meter.h>
#include <opentelemetry/sdk/metrics/meter_provider.h>
#include <opentelemetry/sdk/metrics/meter_provider_factory.h>
```

and compile with either Bazel or CMake, I see the error messages pasted below

My Makefile:
```
build: download-src
	${MKDIR_P} build
	cd build && \
	${CMAKE} -DCMAKE_INSTALL_PREFIX=../installed \
		-DCMAKE_CXX_STANDARD=17 \
		-DWITH_STL=CXX17 \
		-DWITH_ABSEIL=ON \
		-Dabsl_ROOT=../../abseil/installed \
		-DBUILD_TESTING=OFF \
		../src
```

Bazel targets includes the following targets in the dependency
```
        "@io_opentelemetry_cpp//api",
        "@io_opentelemetry_cpp//exporters/ostream:ostream_metric_exporter",
        "@io_opentelemetry_cpp//exporters/ostream:ostream_span_exporter",
        "@io_opentelemetry_cpp//sdk:headers",
        "@io_opentelemetry_cpp//sdk/src/metrics",
        "@io_opentelemetry_cpp//sdk/src/trace",
        "@opentelemetry-protos//:otel_exporter_protos",
```

**What is the expected behavior?**
That it compiles.

**What is the actual behavior?**
What did you see instead?

```
bazel-out/aarch64-fastbuild/bin/external/io_opentelemetry_cpp/sdk/_virtual_includes/headers/opentelemetry/sdk/metrics/data/exemplar_data.h:29:53: error: 'SpanContext' is not a member of 'opentelemetry::v1::sdk::trace' 29 | static ExemplarData Create(std::shared_ptr<trace::SpanContext> context, | ^~~~~~~~~~~ bazel-out/aarch64-fastbuild/bin/external/io_opentelemetry_cpp/sdk/_virtual_includes/headers/opentelemetry/sdk/metrics/data/exemplar_data.h:29:64: error: template argument 1 is invalid 29 | static ExemplarData Create(std::shared_ptr<trace::SpanContext> context, | ^ bazel-out/aarch64-fastbuild/bin/external/io_opentelemetry_cpp/sdk/_virtual_includes/headers/opentelemetry/sdk/metrics/data/exemplar_data.h:50:16: error: 'SpanContext' in namespace 'opentelemetry::v1::sdk::trace' does not name a type 50 | const trace::SpanContext &GetSpanContext() const noexcept { return context_; } | ^~~~~~~~~~~
```

**Additional context**

Compiling with these header files would work properly

```
#include <opentelemetry/exporters/ostream/metric_exporter_factory.h>
#include <opentelemetry/exporters/ostream/span_exporter_factory.h>
#include <opentelemetry/metrics/provider.h>
#include <opentelemetry/sdk/metrics/aggregation/default_aggregation.h>
#include <opentelemetry/sdk/metrics/aggregation/histogram_aggregation.h>
#include <opentelemetry/sdk/metrics/push_metric_exporter.h>
#include <opentelemetry/sdk/metrics/view/instrument_selector_factory.h>
#include <opentelemetry/sdk/metrics/view/meter_selector_factory.h>
#include <opentelemetry/sdk/metrics/view/view_factory.h>
#include <opentelemetry/sdk/trace/exporter.h>
#include <opentelemetry/sdk/trace/processor.h>
#include <opentelemetry/sdk/trace/simple_processor_factory.h>
#include <opentelemetry/sdk/trace/tracer_provider_factory.h>
#include <opentelemetry/trace/provider.h>
#include <opentelemetry/trace/span_context.h>
```

I can even compile with a declaration of SpanContext
```
   std::shared_ptr<opentelemetry::v1::trace::SpanContext> context;
```

--- a/sdk/include/opentelemetry/sdk/metrics/data/exemplar_data.h
+++ b/sdk/include/opentelemetry/sdk/metrics/data/exemplar_data.h
@@ -26,7 +26,7 @@ using MetricAttributes = opentelemetry::sdk::common::OrderedAttributeMap;
 class ExemplarData
 {
 public:
-  static ExemplarData Create(std::shared_ptr<trace::SpanContext> context,
+  static ExemplarData Create(std::shared_ptr<opentelemetry::trace::SpanContext> context,
                              const opentelemetry::common::SystemTimestamp &timestamp,
                              const PointDataAttributes &point_data_attr)
   {
@@ -47,7 +47,7 @@ public:
    * Returns the SpanContext associated with this exemplar. If the exemplar was not recorded
    * inside a sampled trace, the Context will be invalid.
    */
-  const trace::SpanContext &GetSpanContext() const noexcept { return context_; }
+  const opentelemetry::trace::SpanContext &GetSpanContext() const noexcept { return context_; }
 
   static PointType CreateSumPointData(ValueType value)
   {
@@ -68,13 +68,13 @@ public:
   static PointType CreateDropPointData() { return DropPointData{}; }
 
 private:
-  ExemplarData(std::shared_ptr<trace::SpanContext> context,
+  ExemplarData(std::shared_ptr<opentelemetry::trace::SpanContext> context,
                opentelemetry::common::SystemTimestamp timestamp,
                const PointDataAttributes &point_data_attr)
       : context_(*context.get()), timestamp_(timestamp), point_data_attr_(point_data_attr)
   {}
 
-  trace::SpanContext context_;
+  opentelemetry::trace::SpanContext context_;
   opentelemetry::common::SystemTimestamp timestamp_;
   PointDataAttributes point_data_attr_;
 };

--- a/sdk/include/opentelemetry/sdk/metrics/exemplar/reservoir_cell.h
+++ b/sdk/include/opentelemetry/sdk/metrics/exemplar/reservoir_cell.h
@@ -137,13 +137,13 @@ private:
       auto current_ctx = span->GetContext();
       if (current_ctx.IsValid())
       {
-        context_.reset(new trace::SpanContext{current_ctx});
+        context_.reset(new opentelemetry::trace::SpanContext{current_ctx});
       }
     }
   }
 
   // Cell stores either long or double values, but must not store both
-  std::shared_ptr<trace::SpanContext> context_;
+  std::shared_ptr<opentelemetry::trace::SpanContext> context_;
   nostd::variant<int64_t, double> value_;
   opentelemetry::common::SystemTimestamp record_time_;
   MetricAttributes attributes_;

